packages/react/src/components/DashboardEditor/DashboardEditorCardRenderer.jsx

```diff
@@ -164,65 +126,225 @@ const renderCustomCard = (props) => {
+      (e) => {
+        handleOnClick(setSelectedCardId, cardConfig.id);
+        cardConfig.onMouseDown(e);
+      },
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      [cardConfig.id, cardConfig.onMouseDown, setSelectedCardId]
+    );
+    // Need to memoize these card props at this level to improve performance
+    const cardProps = useMemo(
+      () => ({
+        ...cardConfig,
+        style,
+        key: cardConfig.id,
+        tooltip: cardConfig.description,
+        i18n,
+        availableActions,
+        onCardAction: handleOnCardAction,
+        renderIconByName,
+        tabIndex: 0,
+        onKeyDown: handleOnCardKeyDown,
+        onMouseDown: handleOnCardMouseDown,
+        className: `${baseClassName}--preview__card`,
+        isSelected,
+        // Add the show gallery to image card
+        onBrowseClick:
+          cardConfig.type === CARD_TYPES.IMAGE && isNil(cardConfig.content?.src)
+            ? onShowImageGallery
+            : undefined,
+        validateUploadedImage:
+          cardConfig.type === CARD_TYPES.IMAGE ? onValidateUploadedImage : undefined,
+      }),
+      [
+        baseClassName,
+        cardConfig,
+        handleOnCardAction,
+        handleOnCardKeyDown,
+        handleOnCardMouseDown,
+        i18n,
+        isSelected,
+        onShowImageGallery,
+        onValidateUploadedImage,
+        renderIconByName,
+        style,
+      ]
+    );
+
+    useEffect(() => {
+      const originalDynamicHotspot = cardProps.values?.hotspots?.find(
+        (hotspot) => hotspot.type === 'dynamic'
       );
+      // if we have dynamic hotspots and a way to fetch them, fetch them
+      if (originalDynamicHotspot && cardProps.onFetchDynamicDemoHotspots) {
+        cardProps.onFetchDynamicDemoHotspots().then((hotspots) =>
+          setDynamicHotspots(
+            hotspots.map((hotspot) => ({
+              ...hotspot,
+              ...omit(originalDynamicHotspot, 'type', 'x', 'y'),
+            }))
+          )
+        );
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [cardProps.values?.hotspots?.length, cardProps.onFetchDynamicDemoHotspots]);
+
+    if (!isCardJsonValid(cardProps)) {
+      return renderDefaultCard(cardProps);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [others.values?.hotspots?.length, others.onFetchDynamicDemoHotspots]);
-
-  if (!isCardJsonValid(others)) {
-    return renderDefaultCard(others);
-  }
-
-  switch (others.type) {
-    case CARD_TYPES.VALUE:
-      return renderValueCard(others);
-    case CARD_TYPES.TIMESERIES:
-      return renderTimeSeriesCard(others);
-    case CARD_TYPES.BAR:
-      return (
-        <EditorBarChartCard
-          {...others}
-          dataItems={dataItems}
-          availableDimensions={availableDimensions}
-        />
+    // If the renderCardPreview function returns a valid value for the card, use it,
+    if (
+      typeof renderCardPreview === 'function' &&
```

Comment:
should we just remove this? We're not using it and it allows consumers to stray from the design

Reply:
I don't think we can since we published a prop type with it.